src/creature.cpp

```diff
@@ -188,6 +188,9 @@ bool Creature::sees( const tripoint &t, bool is_player ) const
         }
         if( is_player ) {
             // Special case monster -> player visibility, forcing it to be symmetric with player vision.
+            if( g->u.move_mode == "sneak" ) { // not symmetric if the player is sneaking
```

Comment:
Why are you suddenly referring to `g->u`?

Reply:
The is_player flag is set if the target is the player, so i need to refer to the player like that to check if he is sneaking. 
Maybe it would be better to check if the target creature is sneaking in the `bool Creature::sees( const Creature &critter )` function and then adding a parameter to this function like: `bool Creature::sees( const tripoint &t, bool is_player, bool is_sneaking )` 
That would also prepare for sneaking enemies (pretty terrifying thought)

Reply:
A comment explaining the use of the global `g->u` and a `@todo` mark should be fine. You could also make a `virtual bool Creature::is_sneaking() const` function, implemented for the player to check the `move_mode`.